process/logsevents/nftsProcessor.go

```diff
@@ -46,30 +50,44 @@ func (np *nftsProcessor) processEvent(args *argsProcessEvent) (string, bool) {
 		return "", false
 	}
 
-	sender := args.event.GetAddress()
-	if np.shardCoordinator.ComputeId(sender) == np.shardCoordinator.SelfId() {
-		np.processNFTEventOnSender(args.event, args.accounts, args.tokens, args.timestamp, args.tagsCount)
-	}
-
 	// topics contains:
 	// [0] -- token identifier
 	// [1] -- nonce of the NFT (bytes)
-	// [2] -- receiver NFT address -- in case of NFTTransfer OR ESDT token data in case of NFTCreate
+	// [2] -- value
+	// [3] -- receiver NFT address -- in case of NFTTransfer OR ESDT token data in case of NFTCreate
 	topics := args.event.GetTopics()
-	token := string(topics[0])
 	nonceBig := big.NewInt(0).SetBytes(topics[1])
+	if nonceBig.Uint64() == 0 {
+		// is a fungible token should return
+		return "", false
+	}
+
+	sender := args.event.GetAddress()
+	senderShardID := np.shardCoordinator.ComputeId(sender)
+	if senderShardID == np.shardCoordinator.SelfId() {
+		np.processNFTEventOnSender(args.event, args.accounts, args.tokens, args.timestamp, args.tagsCount)
+	}
+
+	token := string(topics[0])
 	identifier := converters.ComputeTokenIdentifier(token, nonceBig.Uint64())
-	shouldReturn := eventIdentifier != core.BuiltInFunctionESDTNFTTransfer || len(topics) < 3
-	if shouldReturn {
+
+	if !np.shouldAddReceiverData(args) {
 		return identifier, true
 	}
 
-	receiver := topics[2]
-	if np.shardCoordinator.ComputeId(receiver) != np.shardCoordinator.SelfId() {
+	valueBig := big.NewInt(0).SetBytes(topics[2])
```

Comment:
maybe we should have a `topicsProcessor` or something like this. This way it will be easier to refer to the receiver or the value via `topicProc.GetValue()` instead of `topics[2]`. 
if the order isn't the same, we can have different implementations for each flow that generates logs

Reply:
added a task for this [EN-10462]